core/kernel/classes/class.Property.php

```diff
@@ -222,6 +226,35 @@ public function setRange(core_kernel_classes_Class $class)
         return (bool) $returnValue;
     }
 
+    public function getDependsOnProperty(): ?core_kernel_classes_Property
+    {
+        if ($this->dependsOnProperty === null) {
+            $dependsOnProperty = $this->getProperty(GenerisRdf::PROPERTY_DEPENDS_ON_PROPERTY);
+            $dependsOnPropertyValue = $this->getOnePropertyValue($dependsOnProperty);
+
+            if (
+                $dependsOnPropertyValue !== null
+                && $dependsOnPropertyValue->getUri() !== GenerisRdf::PROPERTY_DEPENDS_ON_PROPERTY
+            ) {
+                $returnValue = $this->getProperty($dependsOnPropertyValue);
+                $this->dependsOnProperty = $returnValue;
+            }
+        }
+
+        return $this->dependsOnProperty;
+    }
+
+    public function setDependsOnProperty(core_kernel_classes_Property $property): bool
+    {
+        $numberOfUpdatedRows = $this->getImplementation()->setDependsOnProperty($this, $property);
+
+        if ($numberOfUpdatedRows) {
+            $this->dependsOnProperty = $property;
+        }
+
+        return (bool) $numberOfUpdatedRows;
```

Comment:
Do you really think a setter should return number of updated rows? And why is this one inside a setter method?